### PR TITLE
Port planning scene ros api tutorial

### PR DESCRIPTION
**Pull request by vatanaksoytezer**
- Port planning scene ros api tutorial and adresses #8 
- Needs #69 s demo.launch.py to work as expected

### CODE REVIEW

**Comment by vatanaksoytezer** [comment]
Also there seems something wrong with ros2 run moveit_ros_planning moveit_print_planning_scene_info. Which gives me:

[INFO] [1618491970.223003091] [print_planning_scene_info]: Getting planning scene info to print
[INFO] [1618491970.223125481] [moveit_rdf_loader.rdf_loader]: Robot model parameter not found! Did you remap 'robot_description'?
[ERROR] [1618491970.227218950] [moveit_ros.planning_scene_monitor.planning_scene_monitor]: Robot model not loaded
[ERROR] [1618491970.227438829] [print_planning_scene_info]: Planning scene not configured

File: doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp [hunk 1]
```diff
@@ -178,14 +192,14 @@ int main(int argc, char** argv)
...
+  RCLCPP_INFO(LOGGER, "Attaching the object to the hand and removing it from the world.");
   planning_scene.world.collision_objects.clear();
   planning_scene.world.collision_objects.push_back(remove_object);
   planning_scene.robot_state.attached_collision_objects.push_back(attached_object);
   planning_scene.robot_state.is_diff = true;
-  planning_scene_diff_publisher.publish(planning_scene);
+  planning_scene_diff_publisher->publish(planning_scene);
 
-  visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window to continue the demo");
+  prompt("Press 'next' in the RvizVisualToolsGui window to continue the demo");
```
Comment (MarqRazz) [inline doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp:202]: Do you want to change the prompt prints to not include `RvizVisualToolsGui`?

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst [hunk 2]
```diff
@@ -16,19 +11,24 @@ Getting Started
...
+  ros2 launch moveit2_tutorials planning_scene_ros_api_tutorial.launch.py
 
 **Note:** This tutorial uses the **RvizVisualToolsGui** panel to step through the demo. To add this panel to RViz, follow the instructions in the `Visualization Tutorial <../quickstart_in_rviz/quickstart_in_rviz_tutorial.html#rviz-visual-tools>`_.
 
-After a short moment, the RViz window should appear and look similar to the one at the top of this page. To progress through each demo step either press the **Next** button in the **RvizVisualToolsGui** panel at the bottom of the screen or select **Key Tool** in the **Tools** panel at the top of the screen and then press **N** on your keyboard while RViz is focused.
+After a short moment, the RViz window should appear and look similar to the one at the `Visualization Tutorial <../quickstart_in_rviz/quickstart_in_rviz_tutorial.html#rviz-visual-tools>`_. To progress through each demo step either press the **Next** button in the **RvizVisualToolsGui** panel at the bottom of the screen or select **Key Tool** in the **Tools** panel at the top of the screen and then press **N** on your keyboard while RViz is focused.
 
 Expected Output
 ---------------
```
Comment (MarqRazz) [inline doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst:34]: Do we still need this note?
>**Note:** You may see an error message reading :red:`Found empty JointState message`. This is a known bug and will be fixed soon.

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst [hunk 3]
```diff
@@ -16,19 +11,24 @@ Getting Started
...
+  ros2 launch moveit2_tutorials planning_scene_ros_api_tutorial.launch.py
 
 **Note:** This tutorial uses the **RvizVisualToolsGui** panel to step through the demo. To add this panel to RViz, follow the instructions in the `Visualization Tutorial <../quickstart_in_rviz/quickstart_in_rviz_tutorial.html#rviz-visual-tools>`_.
 
-After a short moment, the RViz window should appear and look similar to the one at the top of this page. To progress through each demo step either press the **Next** button in the **RvizVisualToolsGui** panel at the bottom of the screen or select **Key Tool** in the **Tools** panel at the top of the screen and then press **N** on your keyboard while RViz is focused.
+After a short moment, the RViz window should appear and look similar to the one at the `Visualization Tutorial <../quickstart_in_rviz/quickstart_in_rviz_tutorial.html#rviz-visual-tools>`_. To progress through each demo step either press the **Next** button in the **RvizVisualToolsGui** panel at the bottom of the screen or select **Key Tool** in the **Tools** panel at the top of the screen and then press **N** on your keyboard while RViz is focused.
```
Comment (MarqRazz) [inline doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst:31]:
```suggestion
After a short moment, the RViz window should appear and look similar to `Visualization Tutorial <../quickstart_in_rviz/quickstart_in_rviz_tutorial.html#rviz-visual-tools>`_. To progress through each demo step either press the **Next** button in the **RvizVisualToolsGui** panel at the bottom of the screen or select **Key Tool** in the **Tools** panel at the top of the screen and then press **N** on your keyboard while RViz is focused.
```

**Review by MarqRazz — COMMENTED** [review]
This looks good @vatanaksoytezer.

Is there a fix/issue open for getting `moveit_print_planning_scene_info` to work?

File: doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst (hunk 2 above)
Comment (vatanaksoytezer) [reply doc/planning_scene_ros_api/planning_scene_ros_api_tutorial.rst:34]: I don't think so, I will remove it.

**Comment by vatanaksoytezer** [comment]
> Is there a fix/issue open for getting `moveit_print_planning_scene_info` to work?

Yeah, I've talked this with @JafarAbdi , I wrote the todo for it but I forgot to open up the issue. Will open today. He has mentioned he knew what the problem is and will fix it when he has some spare time to work on that.

File: doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp (hunk 1 above)
Comment (vatanaksoytezer) [reply doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp:202]: They were actually not working as intended for me, so I used the normal function one.

Comment (MarqRazz) [reply doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp:202]: The last time I checked rviz_visual_tools GUI had not been ported to ROS2.

In the other tutorials I ported I commented out the `visual_tools.prompt` and added my own to make it easy to re-enable when it gets added back in. It's up to you if you want to do the same approach here...
```
prompt("Press 'Enter' to start the demo");
/* visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window to start the demo"); */
```

Comment (vatanaksoytezer) [reply doc/planning_scene_ros_api/src/planning_scene_ros_api_tutorial.cpp:202]: Yeah sure, I can make it that way.

**Review by MarqRazz — APPROVED** [review]

**Pull request merged**